wondrous-app/components/ProjectOnboarding/BasicsSetup/ProjectLogo.tsx

```diff
@@ -0,0 +1,141 @@
+import { makeUniqueId } from '@apollo/client/utilities';
+import { Grid } from '@mui/material';
+import ImageUpload from 'components/Settings/imageUpload';
+import { InputWrapper } from 'components/TextField/styles';
+import { AVATAR_EDITOR_TYPES } from 'constants/avatarEditor';
+import { useEffect, useRef, useState } from 'react';
+import { useOrgBoard } from 'utils/hooks';
+import { handleImageFile, uploadMedia } from 'utils/media';
+import { ButtonsPanel } from '../Shared';
+import { CONFIG, TYPES } from '../Shared/constants';
+import { PageLabel, TextArea } from '../Shared/styles';
+
+const ProjectLogo = () => {
+  const { setStep, updateOrg, orgData } = useOrgBoard();
+  const [data, setData] = useState({
+    profilePicture: null,
+    headerPicture: null,
+  });
+
+  const handleChange = (value, name) => setData((prev) => ({ ...prev, [name]: value }));
+
+  useEffect(() => {
+    if (data.profilePicture !== orgData?.profilePicture) {
+      setData((prev) => ({ ...prev, profilePicture: orgData?.profilePicture }));
+    }
+    if (data.headerPicture !== orgData?.headerPicture) {
+      setData((prev) => ({ ...prev, headerPicture: orgData?.headerPicture }));
+    }
+  }, [orgData?.profilePicture, orgData?.headerPicture]);
+
+  const ref = useRef(null);
+
+  const FIELDS = [
+    {
+      name: 'profilePicture',
+      label: 'Project Logo',
+      imageType: AVATAR_EDITOR_TYPES.ICON_IMAGE,
+      type: 'image',
+    },
+    {
+      name: 'headerPicture',
+      label: 'Project Cover',
+      imageType: AVATAR_EDITOR_TYPES.HEADER_IMAGE,
+      type: 'image',
+    },
+    {
+      name: 'description',
+      label: 'Bio',
+      type: 'text',
+    },
+  ];
+
+  const transformAndUploadMedia = async ({ file }) => {
+    if (!file) return null;
+
+    const imageFile = handleImageFile({ file, id: makeUniqueId('temp') });
+    await uploadMedia(imageFile);
+    return { ...imageFile };
+  };
+
+  const onClick = async () => {
+    let inputData = {};
+    if (data.profilePicture !== orgData?.profilePicture) {
+      const { filename } = await transformAndUploadMedia({ file: data.profilePicture });
+      inputData = {
+        ...inputData,
+        profilePicture: filename,
+      };
+    }
+    if (ref?.current?.value !== orgData?.description) {
```

Comment:
Is there a reason we're using refs and not state?

Reply:
I wanted to avoid a second state for description. If we add it to the main state object ( the one with profilePic & headerPic ) the description state would update on the onChange event and would trigger flickers on the SafeImage component.